iac-common/src/test/java/org/sonar/iac/common/reports/AbstractExternalRulesDefinitionTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.common.reports;
+
+import org.sonar.api.SonarRuntime;
+
+public class AbstractExternalRulesDefinitionTest extends org.sonar.iac.common.testing.AbstractExternalRulesDefinitionTest {
```

Comment:
Why such test is needed? Maybe it can have another name to not have a "name collision"?
When tried to understand it locally, when I run it from IDE or CMD, then I see that only `externalRepositoryShouldBeInitializedWithSonarRuntime` from base class is executed (the `noOpRulesDefinitionShouldNotDefineAnyRule` is skipped). I don't know why (bug in JUnit?) please investigate it.

Reply:
OK, I found the reason. The `org.sonar.iac.common.reports.AbstractExternalRulesDefinitionTest` overrides the method `noOpRulesDefinitionShouldNotDefineAnyRule()` and doesn't repeat the `@Test` annotation, so the test from the base class was missed in execution.